### DISCUSSION TITLE
Can't setup storybook in a monorepo when installed in the root directory

### DISCUSSION THREAD

**Discussion by HitBox38** [storybookjs/storybook#24814] [Help]
I'm building a monorepo for React TypeScript custom components in Lerna and I want to use Storybook on all of my packages. I installed Storybook in the root directory, but when I try to run my stories I get this error: 
![image](https://github.com/storybookjs/storybook/assets/35326023/2a30a381-643c-455b-9ab0-4f34988d394a)

Am I missing something? Would love any help or guides.

**Comment by valentinpalkovic** [comment]
First of all, Storybook is not meant to be installed in the root directory. We always recommend to install Storybook package-wise in a mono repo. If you still want to continue with this approach, then check the root babelrc file. Is everything configured properly to transpile Typescript files? Storybook picks up your babelrc configuration and applies it to transpile your code. The error message essentially says, that babel is not able to understand the Typescript syntax.

**Reply by HitBox38** [reply]
```json
{
  "sourceType": "unambiguous",
  "presets": [
    [
      "@babel/preset-env",
      {
        "targets": {
          "chrome": 100,
          "safari": 15,
          "firefox": 91
        }
      }
    ],
    "@babel/preset-typescript",
    "@babel/preset-react"
  ],
  "plugins": []
}
```
This is my babelrc file. It doesn't seem like I'm missing something.

**Reply by valentinpalkovic** [reply]
Can you provide a minimal reproduction?

**Reply by HitBox38** [reply]
Ran the following commands:
- `npx lerna init`
- `npm i typescript --save-dev`
- `npx -p @storybook/cli sb init`
  - chose manually the following project type: react_project
  - builder: webpack 5
- `npx lerna create chat-components`
- in `.storybook/main.ts` changed value, `"../stories/*/.stories.@(js|jsx|mjs|ts|tsx)"` in the config.stories array to `"../packages/*/.stories.@(js|jsx|mjs|ts|tsx)"`
- created a simple `Message` component in the chat-components package (in the `lib` folder) with a story file:
```typescript
// Message.tsx
import React from "react";

interface Props {
  content: string;
  timestamp: Date;
  sender: string;
}

const Message = ({ content, sender, timestamp }: Props) => {
  return (
    <div>
      {content}, {sender}, {timestamp.toLocaleTimeString()}
    </div>
  );
};

export default Message;
 ```
 
 ```typescript
// Message.stories.tsx
import type { Meta } from '@storybook/react';

import Message from './index';

const meta = {
  title: 'Chat/Message',
  component: Message,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
} satisfies Meta<typeof Message>;

export default meta;
```

I don't have a git set up yet. If you prefer a git repo I'll open one.
